src/pudl/settings.py

```diff
@@ -689,26 +743,20 @@ def _convert_settings_to_dagster_config(d: dict) -> None:
         if isinstance(v, dict):
             _convert_settings_to_dagster_config(v)
         else:
-            try:
-                d[k] = Field(type(v), default_value=v)
-            except DagsterInvalidDefinitionError:
-                # Dagster config accepts a valid dagster types.
-                # Most of our settings object properties are valid types
-                # except for fields like taxonomy which are the AnyHttpUrl type.
-                d[k] = Field(Any, default_value=v)
+            d[k] = DagsterField(type(v), default_value=v)
```

Comment:
I don't think my doc string for `_convert_settings_to_dagster_config()` is accurate. This function does not create Noneable Dagster fields and instead sets the default value of the partition. Therefore I don't think we need to be setting the partition values to the default if partition is `None` in GenericDatasetSettings.

Reply:
Meaning we should just do:
```py
d[k] = DagsterField(type(v))
```

Reply:
I think we can leave it as is we just need to update the doc string.

Reply:
Okay, I tried to update the docstring to reflect my understanding of what's going on here, let me know if I am wrongo.